lua/astrocommunity/pack/java/java.lua

```diff
@@ -0,0 +1,179 @@
+      -- Add javadgb to required
+      utils.list_insert_unique(opts.ensure_installed, { "javadbg", "javatest" })
+    end,
+  },
+
+  {
+    "mfussenegger/nvim-jdtls",
+    ft = { "java" },
+    init = function()
+      utils.list_insert_unique(astronvim.lsp.skip_setup, "jdtls")
+    end,
+    opts = function(_, opts)
+      -- use this function notation to build some variables
+      local root_markers = { ".git", "mvnw", "gradlew", "pom.xml", "build.gradle", ".project" }
+      local root_dir = require("jdtls.setup").find_root(root_markers)
+
+      -- calculate workspace dir
+      local project_name = vim.fn.fnamemodify(vim.fn.getcwd(), ":p:h:t")
+      local workspace_dir = vim.fn.stdpath "data" .. "/site/java/workspace-root/" .. project_name
+      os.execute("mkdir " .. workspace_dir)
+
+      -- get the mason install path
+      local install_path = require("mason-registry").get_package("jdtls"):get_install_path()
+
+      -- get the current OS
+      local os
+      if vim.fn.has("mac") == 1 then
+        os = "mac"
+      elseif vim.fn.has("unix") == 1 then
+        os = "linux"
+      elseif vim.fn.has("win32") == 1 then
+        os = "win"
+      end
+
+      -- ensure that OS is valid
+      if (not os or os == "") then
+        require("astronvim.utils").notify("jdtls: Could not detect valid OS", vim.log.levels.ERROR)
+      end
+
+      local defaults = {
+        cmd = {
+          "java",
+          "-Declipse.application=org.eclipse.jdt.ls.core.id1",
+          "-Dosgi.bundles.defaultStartLevel=4",
+          "-Declipse.product=org.eclipse.jdt.ls.core.product",
+          "-Dlog.protocol=true",
+          "-Dlog.level=ALL",
+          "-javaagent:" .. install_path .. "/lombok.jar",
+          "-Xms1g",
+          "--add-modules=ALL-SYSTEM",
+          "--add-opens",
+          "java.base/java.util=ALL-UNNAMED",
+          "--add-opens",
+          "java.base/java.lang=ALL-UNNAMED",
+          "-jar",
+          vim.fn.glob(install_path .. "/plugins/org.eclipse.equinox.launcher_*.jar"),
+          "-configuration",
+          install_path .. "/config_" .. os,
+          "-data",
+          workspace_dir,
+        },
+        root_dir = root_dir,
+        settings = {
+          java = {},
+        },
+        init_options = {
+          bundles = {
+            vim.fn.glob(require("mason-registry").get_package("java-debug-adapter"):get_install_path() ..
+              "/extension/server/com.microsoft.java.debug.plugin-*.jar"),
+            -- unpack remaining bundles
+            table.unpack(
+              vim.split(
+                vim.fn.glob(
+                  require("mason-registry").get_package("java-test"):get_install_path()
+                  .. "/extension/server/*.jar"
+                ),
+                "\n", {}))
+          },
+        },
+        handlers = {
+          ["language/status"] = function(_, result)
+            -- print(result)
+          end,
+          ["$/progress"] = function(_, result, ctx)
```

Comment:
```suggestion
          ["$/progress"] = function()
```

Remove unused parameters